# HG changeset patch
# User Mike Miller <mtmiller@octave.org>
# Date 1465419296 25200
#      Wed Jun 08 13:54:56 2016 -0700
# Node ID 6b4f6339df911e59ad9fce54ecfd08c993182338
# Parent  30d4645ecb98b2a41e58f343ba6a2c9454087303
num2hex: Report an error on complex arguments (bug #48184)

* hex2num.cc (Fnum2hex): Report an error on complex arguments for Matlab
compatibility.  Add input validation tests.

diff --git a/libinterp/corefcn/hex2num.cc b/libinterp/corefcn/hex2num.cc
--- a/libinterp/corefcn/hex2num.cc
+++ b/libinterp/corefcn/hex2num.cc
@@ -203,16 +203,19 @@ num2hex (single ([-1, 1, e, Inf]))\n\
 @end group\n\
 @end example\n\
 @seealso{hex2num, hex2dec, dec2hex}\n\
 @end deftypefn")
 {
   if (args.length () != 1)
     print_usage ();
 
+  if (args(0).is_complex_type ())
+    error ("num2hex: N must be real");
+
   octave_value retval;
 
   if (args(0).is_single_type ())
     {
       const FloatColumnVector v (args(0).float_vector_value ());
 
       octave_idx_type nchars = 8;
       octave_idx_type nr = v.numel ();
@@ -280,9 +283,13 @@ num2hex (single ([-1, 1, e, Inf]))\n\
     }
 
   return retval;
 }
 
 /*
 %!assert (num2hex (-2:2), ["c000000000000000";"bff0000000000000";"0000000000000000";"3ff0000000000000";"4000000000000000"])
 %!assert (num2hex (single (-2:2)), ["c0000000";"bf800000";"00000000";"3f800000";"40000000"])
+
+%!error num2hex ()
+%!error num2hex (1,2)
+%!error num2hex (1j)
 */
